### PR TITLE
net: sntp: Handle case of request timeout and fix up sample

### PR DESCRIPTION
**Pull request by pfalcon**
Previously, a case when poll() call timed out wasn't handled, and
recv() was called unconditionally. In the case of timeout, recv()
itself would hang indefinitely.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

### CODE REVIEW

**Comment by pfalcon** [comment]
@galak, Getting SNTP into a workable state is on critical path to get Google IoT SDK working, please help to review/merge this.

File: samples/net/sockets/sntp_client/src/main.c [hunk 1]
```diff
@@ -33,17 +33,20 @@ void main(void)
...
 		       sizeof(struct sockaddr_in));
 	if (rv < 0) {
-		LOG_ERR("Failed to init sntp ctx: %d", rv);
+		LOG_ERR("Failed to init SNTP IPv4 ctx: %d", rv);
 		goto end;
 	}
 
-	rv = sntp_request(&ctx, K_FOREVER, &epoch_time);
+	LOG_INF("Sending SNTP IPv4 request...");
+	rv = sntp_request(&ctx, 4000, &epoch_time);
```
Comment (jukkar) [inline samples/net/sockets/sntp_client/src/main.c]: For timeout, better use `K_SECONDS(4)` instead, preferably via a define.

**Review by jukkar — CHANGES_REQUESTED** [review]
Minor nit pick, looks ok otherwise.

File: samples/net/sockets/sntp_client/src/main.c (hunk 1 above)
Comment (pfalcon) [reply samples/net/sockets/sntp_client/src/main.c]: Used `K_SECONDS(4)`. As these are supposed to be learning samples, using too much defines would only make them more complicated to follow (and yes, there're different delays, e.g. IPv4 case vs IPv6 case one; better just to have them inline).

Comment (jukkar) [reply samples/net/sockets/sntp_client/src/main.c]: > As these are supposed to be learning samples, using too much defines would only make them more complicated to follow

I am disagreeing here. For the novice user reading the code, it is much easier if the parameter says `SNTP_TIMEOUT` than `K_SECONDS(4)`. Not a big issue thou, it is still better now than before.

**Review by jukkar — APPROVED** [review]

**Review by d3zd3z — APPROVED** [review]
The fix seems good. I'm not sure if the config file and logging changes should be in the same commit.

File: samples/net/sockets/sntp_client/src/main.c [hunk 2]
```diff
@@ -57,18 +60,21 @@ void main(void)
 	rv = sntp_init(&ctx, (struct sockaddr *) &addr6,
 		       sizeof(struct sockaddr_in6));
 	if (rv < 0) {
-		LOG_ERR("Failed to init sntp ctx: %d", rv);
+		LOG_ERR("Failed to init SNTP IPv4 ctx: %d", rv);
```
Comment (alexanderwachter) [inline samples/net/sockets/sntp_client/src/main.c]: I think you have a typo here. Should be SNTP IPv6 right?

Comment (jukkar) [reply samples/net/sockets/sntp_client/src/main.c]: Indeed, that is a good catch.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: samples/net/sockets/sntp_client/src/main.c (hunk 2 above)
Comment (pfalcon) [reply samples/net/sockets/sntp_client/src/main.c]: > Should be SNTP IPv6 right?

Fixed, thanks.

**Comment by pfalcon** [comment]
@jukkar: The issue resolved, CI passed.

**Review by jukkar — APPROVED** [review]

**Pull request merged**